lib/sles4sap/console_redirection.pm

```diff
@@ -102,65 +201,70 @@ This allows OpenQA access to command return codes and output for evaulation by s
 
 sub connect_target_to_serial {
     my (%args) = @_;
-    $args{'target_ip'} //= get_required_var('REDIRECT_TARGET_IP');
-    $args{'ssh_user'} //= get_required_var('REDIRECT_TARGET_USER');
+    $args{target_ip} //= get_required_var('REDIRECT_TARGET_IP');
+    $args{ssh_user} //= get_required_var('REDIRECT_TARGET_USER');
 
-    croak "OpenQA variable WORKER_VM_ID undefined. Run 'redirection_init()' first" unless get_var('WORKER_VM_ID');
-    croak "IP address '$args{'target_ip'}' is not valid." unless grep(/^$RE{net}{IPv4}$/, $args{'target_ip'});
+    # Save original value for 'AUTOINST_URL_HOSTNAME', and point requests to localhost
+    # check os-autoinst/testapi.pm host_ip() function to get an idea about inner workings
+    set_var('AUTOINST_URL_HOSTNAME_ORIGINAL', get_var('AUTOINST_URL_HOSTNAME'));
+    set_var('AUTOINST_URL_HOSTNAME', 'localhost');
```

Comment:
is it possible to move the  two set after all the croak?

Reply:
Done